### PR TITLE
feature/19038-treegraph-fill-x-option

### PR DESCRIPTION
**Pull request by Vazonik**
Added the [fillSpace](https://api.highcharts.com/highcharts/series.treegraph.fillSpace) option for the treegraph to fill the whole plot area in X by all visible points. See #19038.

### CODE REVIEW

**Review by TorsteinHonsi — CHANGES_REQUESTED** [review]
Thanks!

* First of all, we need a visual `@sample` to accompany the doclet for the new option.
* In the PR description, add a link to the API option for `fillX`. It does not yet exist, but it will do after release.

Another thing to consider - how does this work with an inverted treegraph?

**Comment by Vazonik** [comment]
@TorsteinHonsi Thanks for the review!
After advice from the team, I decided to rename the option from `fillX` to `fillSpace`.

> Another thing to consider - how does this work with an inverted treegraph?

With both inverted chart and reversed series it seems to work fine. (I found one bug, but it's unrelated to this feature and I reported it here: #19068.)

File: samples/highcharts/series-treegraph/fillspace-option/demo.css
Comment (TorsteinHonsi) [inline samples/highcharts/series-treegraph/fillspace-option/demo.css:1]: Just `series-treegraph/fillspace` will do for the file path. There are lots of options, we don't need `-option` as part of the path.

File: samples/highcharts/series-treegraph/fillspace-option/demo.html [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<script src="https://code.highcharts.com/highcharts.js"></script>
+<script src="https://code.highcharts.com/modules/treemap.js"></script>
+<script src="https://code.highcharts.com/modules/treegraph.js"></script>
+<script src="https://code.highcharts.com/modules/exporting.js"></script>
+<script src="https://code.highcharts.com/modules/accessibility.js"></script>
+
+<div id="container1"></div>
+<div id="container2"></div>
+<button id="toggleCollapseNodeA">Toggle collapse of nodes A</button>
+<button id="toggleCollapseNodeB">Toggle collapse of nodes B</button>
```
Comment (TorsteinHonsi) [inline samples/highcharts/series-treegraph/fillspace-option/demo.html:10]: We use kebab-case for class names and id's in our demos.

**Review by TorsteinHonsi — CHANGES_REQUESTED** [review]
Thanks, nice demo! Just a little nitpicking now.

**Review by TorsteinHonsi — APPROVED** [review]
Thanks! Now only awaiting CI.

**Pull request merged**